backend/src/role.rs

```diff
@@ -19,6 +31,44 @@ pub enum RoleError {
     Unauthorized,
 }
 
+#[get("/<role_id>/questions")]
+pub async fn get_question_ids(
+    role_id: i32,
+    user: User,
+    db: Database,
+) -> Result<Json<QuestionIdsResponse>, Json<QuestionsError>> {
+    // First check that the role is valid and the user should be able to access the ids.
+    // We can't use the helper function below since behaviour depends on the draft
+    // status of the campaign.
+
+    let role = db
+        .run(move |conn| Role::get_from_id(conn, role_id))
+        .await
+        .ok_or(QuestionsError::RoleNotFound)?;
+
+    let campaign = db
+        .run(move |conn| Campaign::get_from_id(conn, role.campaign_id))
+        .await
+        .ok_or(QuestionsError::CampaignNotFound)?;
+
+    let org_user = db
+        .run(move |conn| OrganisationUser::get(conn, campaign.organisation_id, user.id))
+        .await
+        .ok_or(QuestionsError::Unauthorized)?;
+
+    // Prevent people from viewing while it's in draft mode,
+    // unless they have adequate permissions
+    if campaign.draft && !user.superuser && org_user.admin_level == AdminLevel::ReadOnly {
```

Comment:
I think we were planning on throwing this logic in a util fn a couple routes ago, did we get around to that?

Reply:
There is a helper function to check for admin perms but because this depends on the draft status it doesn't work. There could be one that hasn't been merged yet. I could extract this logic out to a helper method? It will probably be used elsewhere at some point.

Reply:
You can probably replace it with

https://github.com/csesoc/chaos/blob/main/backend/src/database/models.rs#L79

which is our helper type for construcitng orfganisation directors/admins

Reply:
Once CHAOS-48 gets merged in, this should be much easier with util